src/util/WalletBackendAdapter.ts

```diff
@@ -188,15 +205,79 @@ export const makeWalletBridgeFromFollowers = (
     marshaller,
   );
 
+  const { notifier: beansOwingNotifier, updater: beansOwingUpdater } =
+    makeNotifierKit<Number | null>(null);
+
   const watchBeansOwing = async () => {
     for await (const { value } of iterateLatest(beansOwingFollower)) {
+      if (isHalted) return;
       beansOwingUpdater.updateState(Number(value));
     }
   };
 
+  // Reads purses from the cosmos bank module. These are not necessarily real
```

Comment:
consider "Infer purses from Cosmos bank module balances" since, as you say, purses can't be read from Cosmos.

Reply:
Done